docker/l2s/src/index.ts

```diff
@@ -0,0 +1,232 @@
+    ...obj,
+    decimals: DEFAULT_DECIMAL,
+    price: ethPrices[0].price,
+  }));
+
+  //  Initialize provider
+  const rpcProvider = new StaticJsonRpcProvider(config.rpcUrl);
+  const multicallContract = new Contract(
+    config.multicallContractAddress,
+    multicallABI,
+    rpcProvider
+  );
+  const CONTRACT_FUNCTION = "balanceOf";
+  const contracts: Contract[] = [];
+  const callInputs: string[][] = [];
+  const ethEscrow: string[] = [];
+
+  escrowContractTokens.forEach((ctr) => {
+    contracts.push(new Contract(ctr.token_address, erc20ABI, rpcProvider));
+    callInputs.push([ctr.address]);
+  });
+  ethEscrowContractTokens.forEach((ctr) => {
+    ethEscrow.push(ctr.address);
+  });
+
+  console.log("Start getting tokens balance by multicall contract");
+  const tx = await getMultipleContractMultipleData(
+    contracts,
+    multicallContract,
+    CONTRACT_FUNCTION,
+    config.blockNumber,
+    callInputs
+  );
+
+  const ethMulticallResult = await getMultipleAddressEthBalance(
+    ethEscrow,
+    config.blockNumber,
+    multicallContract
+  );
+
+  console.log("Get tokens balance by multicall contract success");
+
+  // format the balance, store values in database
+  const escrowBalances: EthProjectEscrowValue[] = tx.map((balance, i) => {
+    if (balance === undefined) {
+      return {
+        project_id: escrowContractTokens[i].project_id,
+        date: priceTimestamp,
+        address: escrowContractTokens[i].address,
+        token_address: escrowContractTokens[i].token_address,
+        balance: 0,
+        price: 0,
+      };
+    }
+    const tokenDecimal = escrowContractTokens[i].decimals;
+    const balanceFormatted = ethers.formatUnits(
+      balance.toString(),
+      tokenDecimal ? +tokenDecimal : DEFAULT_DECIMAL
+    );
+
+    return {
+      project_id: escrowContractTokens[i].project_id,
+      date: priceTimestamp,
+      address: escrowContractTokens[i].address,
+      token_address: escrowContractTokens[i].token_address,
+      balance: +balanceFormatted,
+      price: priceMap[escrowContractTokens[i].token_address] ?? 0,
+    };
+  });
+
+  const ethEscrowValues: EthProjectEscrowValue[] = ethMulticallResult.map(
+    (balance, i) => {
+      const balanceFormatted = ethers.formatUnits(balance.toString(), "ether");
+      return {
+        project_id: ethEscrowContractTokens[i].project_id,
+        date: priceTimestamp,
+        address: ethEscrowContractTokens[i].address,
+        token_address: ethEscrowContractTokens[i].token_address,
+        balance: +balanceFormatted,
+        price: ethEscrowContractTokens[i].price ?? 0,
+      };
+    }
+  );
+
+  // insert value to Postgres
+  const insertPromises = [
+    ethProjectEscrowTimeseriesValueRepo.insert(escrowBalances),
+    ethProjectEscrowTimeseriesValueRepo.insert(ethEscrowValues),
+  ];
+  await Promise.all(insertPromises);
+  console.log("Insert escrow values to Postgres success. End!");
+};
+
+main().catch(console.log);
```

Comment:
@viettiennguyen029 catch exception if RPC error or something, then wait a time and try index again